core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -1451,6 +1451,144 @@ object ZIOSpec extends ZIOBaseSpec {
           fibs <- ZIO.children
         } yield assert(fibs, isEmpty)
       },
+      testM("daemon fiber race interruption") {
+        def plus1(ref: Ref[Int], latch: Promise[Nothing, Unit]) =
+          latch.succeed(()) *> clock.sleep(200.milliseconds) *> ref.update(_ + 1)
```

Comment:
The sleep could cause flakiness here. Actually you don't really need this `ref`, you could simply do `latch.succeed(()) *> ZIO.never` so that it wouldn't finish.